src/coreclr/vm/jitinterfacegen.cpp

```diff
@@ -38,7 +38,15 @@ void InitJITAllocationHelpers()
             SetJitHelperFunction(CORINFO_HELP_NEWARR_1_ALIGN8, RhpNewArrayFastAlign8);
 #endif
 
-            ECall::DynamicallyAssignFCallImpl(GetEEFuncEntryPoint(RhNewString), ECall::FastAllocateString);
+            PCODE pfnHelper = GetEEFuncEntryPoint(RhNewString);
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+            AllocMemHolder<PortableEntryPoint> portableEntryPoint;
+            pfnHelper = PortableEntryPoint::CreateFromNativeEntryPoint(pfnHelper, &portableEntryPoint);
```

Comment:
We should not need to allocate PortableEntryPoint for FCalls. The FCall's MethodDesc has PortableEntryPoint that we just need to backpatch.

I assume that you ended up doing this since it is ambiguous whether ECall::GetFCallImpl return value is actual code or PortableEntryPoint entrypoint. We should fix this problem by returning extra `bool* pfHasManagedImplementation` from ECall::GetFCallImpl, return true in *pfHasManagedImplementation for the delegate ctor and string ctor fcalls (check whether the dynamic ID is in specific range), and then doing the DoPrestub thing only when *pfHasManagedImplementation is true.